components/camel-swagger-java/src/main/java/org/apache/camel/swagger/RestSwaggerSupport.java

```diff
@@ -194,7 +200,7 @@ public void initSwagger(BeanConfig swaggerConfig, Map<String, Object> config) {
     }
 
     public void renderResourceListing(RestApiResponseAdapter response, BeanConfig swaggerConfig, String contextId, String route, boolean json, boolean yaml,
-                                      ClassResolver classResolver, RestConfiguration configuration) throws Exception {
+    		Exchange exchange, ClassResolver classResolver, RestConfiguration configuration) throws Exception {
```

Comment:
Perhaps instead of `Exchange` we could pass in a `Map<String, String>` of headers?